Mojolicious 9 support

toratora opened the pull request:
As noted in the [Mojolicious 9.0 announcement](https://mojolicious.io/blog/2021/02/14/announcing-mojolicious-9-0/) there are some changes to `Mojolicious::Routes::Route`

> * Cleaned up lesser-used router methods
>    * Removed `detour`, `route` (use `any`)
>    * Renamed over to `requires`, `via` to `methods`

I updated the code to check for the old methods and use them, otherwise use the renamed, non-deprecated methods to make it backwards compatible.

Commit:
use renamed, non-depracated router methods for Mojolicious >= 8.67 support

Commit:
remove extra whitespace that was accidentally added

jberger commented:
I came here to open a bug for this too, thanks @toratora for beating me to it.

FWIW depending on how far back you want to support you can just use `any` rather than `route`, it should work back quite a long ways. Not exactly sure how far. For `requires` yes that needs this type of check 👍

toratora commented:
Thanks, @jberger . Love your work!

@kwakwaversal , you're welcome to skip the `route` method check and just use `any` if you feel that support as far back as 5.27 is sufficient.

jberger commented:
> Thanks, @jberger . Love your work!
/me blushes

> @kwakwaversal , you're welcome to skip the `route` method check and just use `any` if you feel that support as far back as 5.27 is sufficient.
It likely works before Mojolicious 5.27 too but _this_ module requires 5.27 at which point `any` works fine.

kwakwaversal committed:
Update Restify.pm

kwakwaversal committed:
Use `any` over `route` for all versions

Following Mojolicious v9 update to remove `route`, replaces with `any` method that @jberger confirmed works for all version of this plugin.

kwakwaversal commented:
Thanks both, and thanks for making the MR @toratora. I have taken suggestions from the discussion and merged in https://github.com/kwakwaversal/mojolicious-plugin-restify/pull/16 (I was going to use this branch but got in muddle after editing direct on github...).

The newest version should appear on metacpan soon.

kwakwaversal commented:
Out of interest @jberger is:

![image](https://user-images.githubusercontent.com/1439360/110258504-cf078600-7f9a-11eb-8d86-443e9709ae85.png)

Due to support being dropped for older versions of Perl? As in I should drop them from the Perl test matrix?

jberger commented:
Yes, Mojolicious requires 5.16 these days. I'd drop them

kwakwaversal commented:
Hah, thanks. I removed it in the other branch I created. :)

Travis now passing, cheers again.
